bundles/binding/org.openhab.binding.panstamp/src/main/java/org/openhab/binding/panstamp/internal/PanStampGenericBindingProvider.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.panstamp.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.openhab.binding.panstamp.PanStampBindingConfig;
+import org.openhab.binding.panstamp.PanStampBindingProvider;
+import org.openhab.core.items.Item;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author Gideon le Grange
+ * @since 1.8.0
+ */
+public class PanStampGenericBindingProvider extends
+		AbstractGenericBindingProvider implements PanStampBindingProvider {
+
+	private static final Logger logger = LoggerFactory.getLogger(PanStampGenericBindingProvider.class);
+
+	@Override
+	public PanStampBindingConfig<?> getConfig(String itemName) {
+		return (PanStampBindingConfig<?>) bindingConfigs.get(itemName);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		logger.debug("getBindingType");
+		return "panstamp";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+		logger.debug("validateItemType(item=" + item + ",bindingConfig="
+				+ bindingConfig);
+		// can't do much cause can't assume we have a running panStamp network so we can't
+		// check item against endpoint for compatibility
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item,
+			String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
```

Comment:
move super call to end of method, so that any preceding exception thrown will skip superclass's processing of bad bindingConfig